pkg/util/webhooks/webhooks.go

```diff
@@ -185,3 +186,27 @@ func GetVMFromAdmissionReview(ar *admissionv1.AdmissionReview) (new *v12.Virtual
 
 	return &newVM, nil, nil
 }
+
+func GetInstanceTypeSpecFromAdmissionRequest(request *admissionv1.AdmissionRequest) (new *instancetypev1beta1.VirtualMachineInstancetypeSpec, old *instancetypev1beta1.VirtualMachineInstancetypeSpec, err error) {
+
+	raw := request.Object.Raw
+	instancetypeObj := instancetypev1beta1.VirtualMachineInstancetype{}
```

Comment:
What about using `request.Object.GetObjectKind()` or `request.RequestKind` to distinguish between `VirtualMachineInstancetype` and `VirtualMachineClusterInstancetype`? It works like this currently because the CRDs are practically identical. But I wonder if this is the cleanest way to do it?

Reply:
I'm not sure... 
What I mean is that the initial object has almost no importance as long as the `.Spec` is identical in both objects and returns `VirtualMachineInstancetypeSpec`. 
The better approach is that each object will have it's own separate Admit() and validation. But right know, these two objects are being treated similarly. It will make sense to change the object in this function when there will be a distinct difference between the objects.
If you insist I'll add a type switch, but I don't see a reason for this right now.

Reply:
Ok, let's leave it like that for now.